lib/AST/NameLookup.cpp

```diff
@@ -1297,29 +1275,17 @@ NominalTypeDecl::lookupDirect(DeclName name,
 
   if (!useNamedLazyMemberLoading) {
     updateLookupTable(LookupTable);
-  }
-
-  // Look for a declaration with this name.
-  if (auto lookup = tryCacheLookup(LookupTable, name))
-    return lookup.getValue();
-
-  if (!useNamedLazyMemberLoading) {
-    return { };
-  }
-
-  // If we get here, we had a cache-miss and _are_ using
-  // NamedLazyMemberLoading. Try to populate a _single_ entry in the
-  // MemberLookupTable from both this nominal and all of its extensions, and
-  // retry.
-  auto &Table = *LookupTable;
-  if (populateLookupTableEntryFromLazyIDCLoader(ctx, Table,
-                                                name.getBaseName(), this)) {
-    updateLookupTable(LookupTable);
   } else {
-    populateLookupTableEntryFromExtensions(ctx, Table, this,
-                                           name.getBaseName());
+    if (populateLookupTableEntryFromLazyIDCLoader(ctx, *LookupTable,
```

Comment:
I'm a little nervous about doing this on _every_ name lookup. What do you think of my idea to mark lookup tables as "complete" once we've done the populateLookupTableEntryFromLazyIDCLoader() / populateLookupTableEntryFromExtensions() dance once?

Reply:
I’ll push a second PR built on this that gives that a go and we’ll compare perf numbers.

Reply:
Done.  See #28922